src/test/java/org/jenkinsci/plugins/scriptsecurity/scripts/ScriptApprovalTest.java

```diff
@@ -179,4 +178,19 @@ public boolean handleConfirm(Page page, String message) {
         assertNotNull(page.getElementById(String.format("pcp-%s", CLASSPATH_HASH5)));
     }
 
+    private void clickAndWait(HtmlElement e) throws IOException {
+        e.click();
+        e.getPage().getWebClient().waitForBackgroundJavaScript(10000);
+    }
+
+    private void click(HtmlPage page, String id, String value) throws IOException {
+        for (HtmlElement e : page.getElementById(id).getElementsByTagName("button")) {
+            if (e.hasAttribute("class") && value.equals(e.getAttribute("class"))) {
```

Comment:
Yuck. Would it be easier to test if the script approval page just added some more `id`s?